RELEASES.md

```diff
@@ -1,3 +1,7 @@
+# Release rayon-core 1.9.3 (2022-05-13)
```

Comment:
I've set the release date for this Friday, so affected folks have a little time to test it.